### PR TITLE
Set link state up for target ports of SriovNetworkNodePolicy for kubernetes cluster

### LINKED ISSUE #45: Applied SriovNetworkNodePolicy does no bring ETH/IB port with DOWN link state to link state UP
After applying SriovNetworkNodePolicy on vanilla K8s cluster selected ETH port does not change link state from DOWN to UP.

Policy example:

```
apiVersion: sriovnetwork.openshift.io/v1
kind: SriovNetworkNodePolicy
metadata:
  name: policy-1
  namespace: sriov-network-operator
spec:
  nodeSelector:
    feature.node.kubernetes.io/network-sriov.capable: "true"
  resourceName: mlnx2f0
  priority: 99
  mtu: 9000
  numVfs: 8
  nicSelector:
    vendor: "15b3"
    pfNames: [ "ens2f0" ]
  isRdma: true
  deviceType: netdevice         
```

Interface link state after policy applying:

```
4: ens2f0: <BROADCAST,MULTICAST> mtu 9000 qdisc mq state DOWN mode DEFAULT group default qlen 1000
    link/ether 0c:42:a1:2b:74:ae brd ff:ff:ff:ff:ff:ff
    vf 0     link/ether 76:ed:fd:cf:eb:c9 brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state auto, trust off, query_rss off
    vf 1     link/ether 16:07:b5:d3:52:75 brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state auto, trust off, query_rss off
    vf 2     link/ether 02:40:7d:5e:5f:af brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state auto, trust off, query_rss off
    vf 3     link/ether 3a:65:ce:66:a0:24 brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state auto, trust off, query_rss off
    vf 4     link/ether 2a:e4:ae:39:5e:1a brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state auto, trust off, query_rss off
    vf 5     link/ether 0a:58:53:bf:74:e4 brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state auto, trust off, query_rss off
    vf 6     link/ether 12:15:81:9e:d3:c4 brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state auto, trust off, query_rss off
    vf 7     link/ether 2e:9c:10:5b:99:15 brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state auto, trust off, query_rss off

```

My setup info:

1. OS - Ubuntu 20.04.1 with kernel 5.4.0-59-generic
2. K8s cluster - v1.19.5 with Calico management plugin
2. Container runtime - CRI-O v1.19
3. Multus - v3.6 (CNI spec 0.3.1)
4. SRIOV network operator (github master)

### PR DESCRIPTION
**Pull request by Mmduh-483**
fixes #45

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -315,6 +315,17 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 			}
 		}
 	}
+	// Set PF link up for kubernetes cluster
+	if ClusterType == ClusterTypeKubernetes && ifaceStatus.Name != "" {
+		pfLink, err := netlink.LinkByName(ifaceStatus.Name)
+		if err != nil {
```
Comment (moshe010) [inline pkg/utils/utils.go:321]: you can check if the pfLink is not up with 
pfLink.Attrs().OperState == OperUp or something like that. 

Also in what case ifaceStatus.Name will be ""?

**Review by moshe010 — CHANGES_REQUESTED** [review]

File: pkg/utils/utils.go (hunk 1 above)
Comment (Mmduh-483) [reply pkg/utils/utils.go:321]: If some modules are not loaded like ipoib for IB interfaces

Comment (Mmduh-483) [reply pkg/utils/utils.go:321]: Added condition for state and removed the condition for name

**Review by moshe010 — APPROVED** [review]

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -315,6 +315,19 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 			}
 		}
 	}
+	// Set PF link up for kubernetes cluster
+	if ClusterType == ClusterTypeKubernetes {
```
Comment (zshi-redhat) [inline pkg/utils/utils.go:319]: Is there a reason this is special for k8s cluster?

Comment (moshe010) [reply pkg/utils/utils.go:319]: So we encounter this is k8s vanilla? we can remove this check if you think it make sense openshift.

Comment (zshi-redhat) [reply pkg/utils/utils.go:319]: Thanks Moshe!
I think it's fine to enable it for both cluster types, as long as there is numvfs configured for that device, link state will be set to up.
/cc @pliurh

Comment (pliurh) [reply pkg/utils/utils.go]: In openshift, the NetworkManager will do the job. Shall we check whether the link-state first before setting it to up?

Comment (Mmduh-483) [reply pkg/utils/utils.go]: There is a check for the state `if pfLink.Attrs().OperState != netlink.OperUp `

**Pull request merged**